File: src/main/java/com/vspk/bookrest/config/MailConfig.java
package com.vspk.bookrest.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.JavaMailSenderImpl;

import java.util.Properties;

@Configuration
public class MailConfig {
    @Value("${spring.mail.host}")
    private String host;

    @Value("${spring.mail.username}")
    private String username;

    @Value("${spring.mail.password}")
    private String password;

    @Value("${spring.mail.port}")
    private int port;

//    @Value("${spring.mail.protocol}")
//    private String protocol;

    @Bean
    public JavaMailSender getJavaMailSender() {
        var mailSender = new JavaMailSenderImpl();

        mailSender.setHost(host);
        mailSender.setPort(port);
        //mailSender.setProtocol(protocol);
        mailSender.setUsername(username);
        mailSender.setPassword(password);


        var properties = mailSender.getJavaMailProperties();

        properties.setProperty("mail.smtp.auth", "true");
        properties.setProperty("mail.smtp.starttls.enable", "true");
        properties.put("mail.transport.protocol", "smtp");
        properties.put("mail.smtp.ssl.enable", "true");
        properties.put("mail.debug", "true");

        return mailSender;
    }
}

File: src/main/java/com/vspk/bookrest/config/MailConfig.kt
package com.vspk.bookrest.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.JavaMailSenderImpl

@Configuration
open class MailConfig {
    @Value("{spring.mail.host}")
    private lateinit var host: String

    @Value("{spring.mail.username}")
    private lateinit var username: String

    @Value("{spring.mail.password}")
    private lateinit var password: String

    @Value("\${spring.mail.port}")
    private lateinit var port: String

    @Bean
    open fun javaMailSender(): JavaMailSender {
        val mailSender = JavaMailSenderImpl()
        mailSender.host = host
        mailSender.port = port.toInt()
        mailSender.username = username
        mailSender.password = password
        val properties = mailSender.javaMailProperties
        properties.setProperty("mail.smtp.auth", "true")
        properties.setProperty("mail.smtp.starttls.enable", "true")
        properties["mail.transport.protocol"] = "smtp"
        properties["mail.smtp.ssl.enable"] = "true"
        properties["mail.debug"] = "true"
        return mailSender
    }
}